test/pkg.jl

```diff
@@ -139,6 +139,11 @@ temp_pkg_dir() do
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
 
+    Pkg.checkout("Example", "test-branch") #17364
```

Comment:
Bump. What if we make a local clone of Example.jl and create the branch locally from this known commit, and do checkout using the local remote. I don't want to have a test that would start to fail if someone commits anything to a particular branch on github.